lib/express-handlebars.js

```diff
@@ -327,6 +327,24 @@ ExpressHandlebars.prototype._getTemplateName = function (filePath, namespace) {
     return name;
 };
 
+ExpressHandlebars.prototype._getView = function(viewsPath, viewPath) {
+  if (!Array.isArray(viewsPath)) {
+     return this._getTemplateName(path.relative(viewsPath, viewPath));
+  }
+  
+  var self = this;
+  var stripFilename = /\/[^\/\\]+?$/;
+  
+  // Stripping filename (rather than indexOf) in case there are
+  // multiple views folders in the same parent directory
```

Comment:
Yeah, this is why we need a criteria for determine the best match. Should we go with, the file must exist in the dir, and have the shortest length to be the winner?